bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/InterleavedLedgerStorage.java

```diff
@@ -643,4 +655,57 @@ public void onRotateEntryLog() {
     public List<GarbageCollectionStatus> getGarbageCollectionStatus() {
         return Collections.singletonList(gcThread.getGarbageCollectionStatus());
     }
+
+    @Override
+    public OfLong getListOfEntriesOfLedger(long ledgerId) throws IOException {
+        Iterator<LedgerCache.PageEntries> pageEntriesIteratorNonFinal = null;
+        try {
+            pageEntriesIteratorNonFinal = ledgerCache.listEntries(ledgerId).iterator();
+        } catch (Bookie.NoLedgerException noLedgerException) {
+            pageEntriesIteratorNonFinal = Collections.emptyIterator();
+        }
+        final Iterator<LedgerCache.PageEntries> pageEntriesIterator = pageEntriesIteratorNonFinal;
+        return new OfLong() {
```

Comment:
It might be worth moving this class to a separate class so it is easy to write tests for it.

Reply:
moved it to LedgerCache class